pandas/tests/plotting/test_misc.py

```diff
@@ -425,3 +425,53 @@ def test_dictionary_color(self):
         ax = df1.plot(kind="line", color=dic_color)
         colors = [rect.get_color() for rect in ax.get_lines()[0:2]]
         assert all(color == expected[index] for index, color in enumerate(colors))
+
+    def test_has_externally_shared_axis(self):
```

Comment:
Could you make a test that is not testing this internal function itself but the impact of this function brings to plotting api?

Reply:
Okay. I've added a separate function  that imnplements a test based upon the sample code from the bug report which checks that the API call exhibits the desired behaviour..